ext/gmp/gmp.c

```diff
@@ -429,9 +429,7 @@ static int gmp_compare(zval *op1, zval *op2) /* {{{ */
 	zval result;
 
 	gmp_cmp(&result, op1, op2);
-	if (Z_TYPE(result) == IS_FALSE) {
-		return 1;
-	}
+	/* result can only be a zend_long if gmp_cmp hasn't thrown an Error */
```

Comment:
Shouldn't this for an EG(exception) check then? Z_LVAL(result) may not be set if an exception has been thrown.

Reply:
Probably should, but I initially dropped this as I was expecting some kind of memory read error or segfault but that didn't happen, will update this.

Reply:
Please add a `ZEND_ASSERT(Z_TYPE(result) == IS_LONG);` here.